verifying/verifying.go

```diff
@@ -59,9 +59,35 @@ func VerifyVRFNonce(nonce *uint64, m *shared.VRFNonceMetadata, opts ...OptionFun
 	return nil
 }
 
+type ProofVerifier struct {
+	inner *postrs.Verifier
+}
+
+// NewProofVerifier creates a new proof verifier.
+// If powFlags is nil, the recommended PoW flags will be used.
+// The verifier must be closed after use with Close().
+func NewProofVerifier(powFlags *config.PowFlags) (*ProofVerifier, error) {
```

Comment:
I think we should keep the functional options for optional parameters and not use `nil` as a placeholder for "default parameter".

We are already using them in the `proving` package and adding them here in the `verifying` package would make a more consistent API.